temporal-test-server/build.gradle

```diff
@@ -99,28 +92,43 @@ idea {
 }
 
 
-graal {
-  outputName "temporal-test-server"
-  mainClass application.getMainClass().get()
-  javaVersion '17'
-  graalVersion '22.3.1'
-
-  // Don't fallback to running a JVM
-  option "--no-fallback"
-
-  // Signal handling so that ^C actually stops the process
-  option "--install-exit-handlers"
-
-  // If we're on linux, static link everything but libc. Otherwise link
-  // everything dynamically (note the '-' rather than '+' in fromt of
-  // StaticExecutable)
-  option isLinux()
-    ? "-H:+StaticExecutableWithDynamicLibC"
-    : "-H:-StaticExecutable"
-}
-
 def isLinux() {
-  return System.properties['os.name'].toLowerCase().contains('linux')
+    return System.properties['os.name'].toLowerCase().contains('linux')
 }
 
-tasks.build.dependsOn('nativeImage')
+// The graalvm plugin requires we build with Java 11
+if (project.hasProperty("nativeBuild")) {
+    apply plugin: 'org.graalvm.buildtools.native'
+
+    graalvmNative {
+        toolchainDetection = true
+        agent {
+            enabled = true
+            defaultMode = "standard"
+            metadataCopy {
+                outputDirectories.add("src/main/resources/META-INF/native-image/io.temporal/temporal-test-server")
+                mergeWithExisting = false
+            }
+
+        }
+        binaries {
+            main {
+                mainClass = application.getMainClass().get()
+                sharedLibrary = false
+                // Signal handling so that ^C actually stops the process
+                buildArgs.add("--install-exit-handlers")
+                // If we're on linux, static link everything but libc. Otherwise link
+                // everything dynamically (note the '-' rather than '+' in front of
+                // StaticExecutable)
+                buildArgs.add(isLinux() ? "-H:+StaticExecutableWithDynamicLibC": "-H:-StaticExecutable")
+                buildArgs.add("-H:+UnlockExperimentalVMOptions")
+                buildArgs.add("-O4")
+
+                runtimeArgs.add("7233")
```

Comment:
Can you help me understand this a bit? Does it mean that this is the default arg if not provided? https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html is not very clear to me.

Reply:
It sets the arg for `./gradlew -Pagent -PnativeBuild :temporal-test-server:nativeRun` since there isn't another way I could find